shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -436,6 +441,20 @@ - (nullable id)accessibilityElementAtIndex:(NSInteger)index {
     return _semanticsObject;
   }
   SemanticsObject* child = [_semanticsObject children][index - 1];
+
+  // This 'if' block handles adding accessibility support for the embedded platform view.
+  // We first check if the child is a semantic node for a platform view.
+  // If so, we add the platform view as accessibilityElements of the child.
+  shell::FlutterPlatformViewsController* flutterPlatformViewsController =
+      _bridge.get()->flutter_platform_views_controller();
+  if (child.node.platformViewId > -1 && flutterPlatformViewsController) {
```

Comment:
> Is it even possible that the child object's platformViewId got changed but the child still exists in the tree?

The entire a11y system is build on the assumption that SemanticsNodes can easily switch roles. While our current impl of PlatformViews may guarantee that platformViewId on a node don't change (I haven't checked this), somebody else could implement their own PlatformView where switching is possible and power it with our semantics layer.

Reply:
Added the code to clean up the a11y elements.